strings-csharp/ReadStringFromResourceFile/ReadStringFromResourceFile/Classes/ResourceItem.cs

```diff
@@ -1,15 +1,22 @@
 ﻿namespace ReadStringFromResourceFile;
 public class ResourceItem<T>
 {
-    private T? item;
+    private readonly string? _errorMessage;
+    private readonly T? _item;
 
-    public ResourceItem(T? item = default(T))
+    public ResourceItem(T? item = default)
```

Comment:
With the addition of the `errorMessage` constructor, maybe it makes sense to not have a default value here and to make calling one of the two constructors a requirement. I am assuming the default was here to indicate the invalid state, but now we have an explicit error state constructor.